api/search/views.py

```diff
@@ -0,0 +1,176 @@
+from django.shortcuts import render
+
+# Create your views here.
+# desired operation below
+# m -> x?
+# d1 -> a b
+# d2 -> c
+# d3 ->d e
+#
+#
+#
+# 1. x
+# 2. x a
+# 3. x a c
+# 4. x a c d
+# 5. x a c e
+# 6. x b
+# 7. x b c
+# 8. x b c d
+# 9. x b c e
+import json
+from django.http import HttpResponse
+
+from api.mainquery.views import Dimension
+from api.search.models import AnnotatedArticle
+
+STATIC_JSON = """
+{
+    "dimensions":[
+        {
+                    "keywords": [
+                        "a","b","c","d"
+                    ]
+                },
+        {
+                    "keywords": [
+                        "e","f"
+                    ]
+                },
+        {
+                    "keywords": [
+                        "g","h"
+                    ]
+                }
+    ]
+}
+"""
+
+STATIC_JSON_2 = """
+{
+    "dimensions":[
+        {
+                    "keywords": [
+                        "a","b"
+                    ]
+                },
+        {
+                    "keywords": [
+                        "c"
+                    ]
+                },
+        {
+                    "keywords": [
+                        "d"
+                    ]
+                },
+        {
+                    "keywords": [
+                        "e"
+                    ]
+                },
+        {
+                    "keywords": [
+                        "f","g"
+                    ]
+                }
+    ]
+}
+"""
+
+STATIC_JSON_3 = """
+{
+    "dimensions":[
+        {
+                    "keywords": [
+                        "bipolar","genetics"
+                    ]
+                },
+        {
+                    "keywords": [
+                        "manic depressive"
+                    ]
+                }
+    ]
+}
+"""
+
+
+def detail(request):
+    dim = []
+    helper = SearchHelper(dim)
+    resp = helper.get_annotations(STATIC_JSON_3)
+    return HttpResponse("%s" % resp)
+
+
+class SearchHelper(object):
```

Comment:
I think this will need to account for the main query string as well but it could be in next iteration if you wish. So, for now it's ok like this as well.